src/orchestrator/mod.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+    error::Result,
+    model::{PodJob, PodResult},
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, future::Future, path::PathBuf};
+/// Options for sourcing compute environment images.
+pub enum ImageKind {
+    /// A published compute environment image in a container registry. Argument formatted as
+    /// `{server.com/}{name}:{tag}`. Server is optional e.g. (`alpine:latest`).
+    Published(String),
+    /// A packaged compute environment of image+tag as a tarball. Argument is the relative path of
+    /// tarball in orchestrator data directory e.g. (`path/to/image.tar.gz`).
+    Tarball(PathBuf),
+}
+/// Available states of a run.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum RunState {
```

Comment:
Should probably rename to just State due to the usage in pod result. Also perhaps move this to utils since it is used by mutiple modules?

Reply:
Also is there no option for queuing? Sometimes orcherstrator might have max jobs already running like in K8 and has to wait their turn

Reply:
Make an issue about either finding a way to deal with it via enum inheratince (don't think there either)
Other resolved